WcaOnRails/app/models/wcif.rb

```diff
@@ -0,0 +1,30 @@
+# frozen_string_literal: true
+
+class Wcif < ApplicationRecord
+  belongs_to :competition
+
+  # See https://github.com/thewca/worldcubeassociation.org/wiki/wcif
+  def json
```

Comment:
(Just discussing things from the perspective of an outside user for now, because I do agree that internally, our method and database fields could use some work, but I don't think it makes sense to talk about those internals until we've agreed on just what this API should do)

> You're calling two different things "the WCIF" above. One is the "base" document you can set through the API. The other is that document with data from the WCA DB merged in.

That is true, but I would call those "implementation details". The API user shouldn't care about how this data is stored. There is the weirdness of `set`ing one thing and `get`ing something different back, which I'll talk about later.

> You can first set the WCIF to one document, and later get the WCIF, and receive a different document. That is really weird!

I actually didn't find this too weird at first. I saw this as form of the [robustness principle](https://en.wikipedia.org/wiki/Robustness_principle): "Be conservative in what you send, be liberal in what you accept". But maybe that's a flawed analogy and I think there might be a simple tweak on what we already have that achieves the "i set A, i expect to get A back" property you desire (I think you suggested that solution when you suggested ignoring or treating conflicts as errors).

> I think you need to treat the uploaded "base" document and the "merged" document as separate entities. Maybe also the part generated from the DB.

Maybe internally, but I still believe this is something we can keep the API user ignorant of.

> How realistic is it that there is a conflict between the uploaded and generated parts? Could that be ignored or treated as an error? To be clear, I'm asking if there really needs to be "destructive" merging?

I'd argue that with the current implementation, what we're doing **is** ignoring conflicts. It's confusing to someone looking under the hood, because you see that we're storing all the data they give us in the database column, but we clobber all that data when they read it back out. From the perspective of an API user, this would look something like this:

1. I tried to set USNats2017's competition.id to JeremyAwesome2017.
2. I read USNats2017's WCIF, and its competition.id is USNats2017. Looks like they just ignored what i did in step 1) above.

An alternative would be to throw an error whenever the user tries to set conflicting information, but certainly the easiest thing to implement is to just ignore whatever they send us.